contracts/Core/StakeManager.sol

```diff
@@ -173,44 +176,61 @@ contract StakeManager is Initializable, StakeStorage, StateManager, Pause, Stake
     /// @param stakerId The Id of staker associated with sRZR which user want to unstake
     /// @param sAmount The Amount in sRZR
     function unstake(uint32 stakerId, uint256 sAmount) external initialized whenNotPaused {
+        Structs.Staker storage staker = stakers[stakerId];
+        uint32 epoch = _getEpoch(epochLength);
+        require(staker.id != 0, "staker.id = 0");
+        require(staker.stake > 0, "Nonpositive stake");
+        require(locks[msg.sender][staker.tokenAddress][LockType.Unstake].amount == 0, "Existing Unstake Lock");
+        require(locks[msg.sender][staker.tokenAddress][LockType.Withdraw].amount == 0, "Existing Withdraw Lock");
+        require(sAmount > 0, "Non-Positive Amount");
+
+        IStakedToken sToken = IStakedToken(staker.tokenAddress);
+        require(sToken.balanceOf(msg.sender) >= sAmount, "Invalid Amount");
+
+        locks[msg.sender][staker.tokenAddress][LockType.Unstake] = Structs.Lock(sAmount, 0, epoch + lockPeriod);
+        emit Unstaked(msg.sender, epoch, stakerId, sAmount, staker.stake, block.timestamp);
+    }
+
+    function initiateWithdraw(uint32 stakerId) external initialized whenNotPaused {
         State currentState = _getState(epochLength);
         require(currentState != State.Propose, "Unstake: NA Propose");
         require(currentState != State.Dispute, "Unstake: NA Dispute");
         uint32 epoch = _getEpoch(epochLength);
 
         Structs.Staker storage staker = stakers[stakerId];
-        require(staker.id != 0, "staker.id = 0");
-        require(staker.stake > 0, "Nonpositive stake");
-        require(locks[msg.sender][staker.tokenAddress].amount == 0, "Existing Lock");
-        require(sAmount > 0, "Non-Positive Amount");
+        Structs.Lock storage lock = locks[msg.sender][staker.tokenAddress][LockType.Unstake];
+        require(staker.id != 0, "staker doesnt exist");
+        require(lock.unlockAfter != 0, "Did not unstake");
+        require(lock.unlockAfter <= epoch, "Withdraw epoch not reached");
+        require(lock.unlockAfter + withdrawInitiationPeriod >= epoch, "Initiation Period Passed"); // Can Use ExtendLock
 
         // slither-disable-next-line reentrancy-events,reentrancy-no-eth
         rewardManager.giveInactivityPenalties(epoch, stakerId);
 
         IStakedToken sToken = IStakedToken(staker.tokenAddress);
-        require(sToken.balanceOf(msg.sender) >= sAmount, "Invalid Amount");
+        require(sToken.balanceOf(msg.sender) >= lock.amount, "locked amount lost");
```

Comment:
how is it possible? defeats purpose

Reply:
hmm we are not taking custody of sRZRs, so one can transfer them 
mistakes purpose is only to declare that you want to withdraw this much sRZRs 
and then in withdraw we convert